server/providers/skills-provider.js

```diff
@@ -0,0 +1,59 @@
+var dataProvider = require('server/providers/data-provider'),
+    skillsCollectionName = 'skills';
+
+var SkillsProvider = function () {
+    function getSkills(callback) {
+        try {
+            dataProvider.get({}, skillsCollectionName, function (err, data) {
+                if (err) {
+                    callback(err);
+                    return;
+                }
+                
+                callback(null, data);
+            });
+
+        } catch (err) {
+            callback(err);
+        }
+    }
+
+    function addSkill(skill, callback) {
+        try {
+            dataProvider.insert(skill, skillsCollectionName, callback);
+
+        } catch (err) {
+            callback(err);
+        }
+    }
+
+    function removeSkill(skillName, callback) {
+        try {
+            dataProvider.remove({ 'name': skillName }, skillsCollectionName, callback);
+        } catch (err) {
+            callback(err);
+        }
+    }
+
+    function upsertSkills(skills, callback) {
+        try {
+            skills.forEach(function (skill) {
+                dataProvider.update({name: skill.name}, skill, skillsCollectionName, function () {

```

Comment:
I think an incorrect behavior might be present here. Foreach would be a Synchronous function, which will trigger N async operations. Before the Async Operations are finished, you will be out of the forEach loop and you'll call the callback(), which might not be the expected behavior. The callback function here is unused, too.